### PR TITLE
change: lineage query visualization experience enhancement

### PR DESCRIPTION
**Pull request by ytlee93**
*Issue #, if available:*

*Description of changes:* 
lineage query visualization experience enhancement, including embedding legend to html file generate by pyvis and adding in arn information when hover on node
image

*Testing done:*
tox -e black-format & tox -e flake8,pylint,twine,black-check,docstyle --parallel all
tox -r -e py39 tests/integ/sagemaker/lineage/test_lineage_visualize.py
tox -e py39 tests/unit/sagemaker/lineage

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/lineage/query.py [hunk 1]
```diff
@@ -300,13 +303,66 @@ def _import_visual_modules(self):
         get_module("pyvis")
         from pyvis.network import Network
         from pyvis.options import Options
+        from IPython.display import IFrame
 
-        return Network, Options
+        get_module("bs4")
+        from bs4 import BeautifulSoup
+
+        return Network, Options, IFrame, BeautifulSoup
```
Comment (Snoyark) [inline src/sagemaker/lineage/query.py:311]: Did these two new imports require additional installations by end customers?

File: src/sagemaker/lineage/query.py [hunk 2]
```diff
@@ -300,13 +303,66 @@ def _import_visual_modules(self):
         get_module("pyvis")
         from pyvis.network import Network
         from pyvis.options import Options
+        from IPython.display import IFrame
 
-        return Network, Options
+        get_module("bs4")
+        from bs4 import BeautifulSoup
```
Comment (jkasiraj) [inline src/sagemaker/lineage/query.py:309]: Is this module already imported via other dependencies or is this a dependency the user will have to install themselves?

File: src/sagemaker/lineage/query.py [hunk 3]
```diff
@@ -325,23 +381,51 @@ def render(self, elements, path="pyvisExample.html"):
             display graph: The interactive visualization is presented as a static HTML file.
 
         """
-        net = self.Network(height="500px", width="100%", notebook=True, directed=True)
+        net = self.Network(height="600px", width="82%", notebook=True, directed=True)
```
Comment (Snoyark) [inline src/sagemaker/lineage/query.py:378]: How were these numbers created? `82%` especially pops out a bit. Does it break/look worse at `80%`?

File: src/sagemaker/lineage/query.py [hunk 4]
```diff
@@ -300,13 +303,66 @@ def _import_visual_modules(self):
...
+    def _add_legend(self, path):
+        """Embed legend to html file generated by pyvis."""
+        f = open(path, "r")
+        content = self.BeautifulSoup(f, "html.parser")
+
+        legend = """
+            <div style="display: inline-block; font-size: 1vw; font-family: verdana;
+                vertical-align: top; padding: 1vw;">
+                <div>
+                    <div style="background-color: rgb(246, 207, 97); width: 1.6vw; height: 1.6vw;
```
Comment (jkasiraj) [inline src/sagemaker/lineage/query.py]: Can we generate each of these divs programmatically based on the config?

i.e. something like

```
getLegendLine(graph_style):
  return f`<div style = "{style.style}; width: ....>
....'
```

and then iterate through all graph_styles to get them all

File: src/sagemaker/lineage/query.py (hunk 1 above)
Comment (ytlee93) [reply src/sagemaker/lineage/query.py:311]: I didn't add BeautifulSoup into the [test requirements ](https://github.com/Snoyark/sagemaker-python-sdk/blob/master/requirements/extras/test_requirements.txt) but it still ran successfully on unit and integ tests, does it mean it doesn't require additional installations by end customers?

File: src/sagemaker/lineage/query.py (hunk 3 above)
Comment (ytlee93) [reply src/sagemaker/lineage/query.py:378]: 82% gives the pyvis graph max area to display, 80% doesn't break or look very different, do I have to change it then?

File: src/sagemaker/lineage/query.py (hunk 2 above)
Comment (ytlee93) [reply src/sagemaker/lineage/query.py:309]: I didn't add BeautifulSoup into the [test requirements ](https://github.com/Snoyark/sagemaker-python-sdk/blob/master/requirements/extras/test_requirements.txt)but it still ran successfully on unit and integ tests, so I think they are installed via other dependencies already?

**Review by jkasiraj — APPROVED** [review]

File: src/sagemaker/lineage/query.py (hunk 3 above)
Comment (Snoyark) [reply src/sagemaker/lineage/query.py:378]: Nope, just curious.

**Review by Snoyark — APPROVED** [review]

**Pull request merged**